collection.go

```diff
@@ -697,6 +741,165 @@ func (cl *collectionLoader) populateDeferredMaps() error {
 	return nil
 }
 
+// translateStructData fills in all fields in `from` that exist in `to` with contents of fromData.
+func translateStructData(from *btf.Struct, fromData []byte, to *btf.Struct) ([]byte, error) {
+	innerName := strings.TrimPrefix(to.Name, structOpsValuePrefix)
+	toData := make([]byte, int(to.Size))
+
+	var inner *btf.Struct
+	var innerOff int
+
+	for _, m := range to.Members {
+		st, ok := btf.As[*btf.Struct](btf.UnderlyingType(m.Type))
+		if ok && st.Name == innerName {
+			inner = st
+			innerOff = int(m.Offset / 8)
+		}
+	}
+
+	if inner == nil {
+		return nil, fmt.Errorf("member %s not found in %s", innerName, to.Name)
+	}
+
+	kernIndexByName := make(map[string]btf.Member, len(inner.Members))
+	for _, m := range inner.Members {
+		kernIndexByName[m.Name] = m
+	}
+
+	for _, m := range from.Members {
+		if m.BitfieldSize > 0 {
+			return nil, fmt.Errorf("bitfield %s not supported", m.Name)
+		}
+
+		kernMember, ok := kernIndexByName[m.Name]
+		if !ok {
+			continue
+		}
+
+		if kernMember.BitfieldSize > 0 {
+			return nil, fmt.Errorf("bitfield %s not supported in kern struct", kernMember.Name)
+		}
+
+		sz, err := btf.Sizeof(m.Type)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve size of %s: %w", m.Name, err)
+		}
+
+		kernSz, err := btf.Sizeof(kernMember.Type)
+		if err != nil {
+			return nil, fmt.Errorf("failed to resolve size of %s: %w", kernMember.Name, err)
+		}
+
+		if sz != kernSz {
```

Comment:
This check is pretty basic. Is that all that libbpf does as well?

Might make sense to call `btf.CheckTypeCompatibility` if libbpf does something similar.